src/System.Drawing.Design.Primitives/src/System/Drawing/Design/IToolboxService.cs

```diff
@@ -0,0 +1,212 @@
+        /// </summary>
+        /// <param name="toolboxItem">tool box item</param>
+        /// <param name="host">host</param>
+        void AddLinkedToolboxItem(ToolboxItem toolboxItem, IDesignerHost host);
+
+        /// <summary>
+        /// Adds a new tool to the toolbox under the specified category.
+        /// </summary>
+        /// <param name="toolboxItem">tool box item</param>
+        /// <param name="category">category</param>
+        /// <param name="host">host</param>
+        void AddLinkedToolboxItem(ToolboxItem toolboxItem, string category, IDesignerHost host);
+
+        /// <summary>
+        /// Adds a new tool to the toolbox under the default category.
+        /// </summary>
+        /// <param name="toolboxItem">tool box item</param>
+        void AddToolboxItem(ToolboxItem toolboxItem);
+
+        /// <summary>
+        /// Adds a new tool to the toolbox under the specified category.
+        /// </summary>
+        /// <param name="toolboxItem">tool box item</param>
+        /// <param name="category">category</param>
+        void AddToolboxItem(ToolboxItem toolboxItem, string category);
+
+        /// <summary>
+        /// Gets a toolbox item from a previously serialized object.
+        /// </summary>
+        /// <param name="serializedObject">serialized Object</param>
+        /// <returns>toolbox item</returns>
+        ToolboxItem DeserializeToolboxItem(object serializedObject);
+
+        /// <summary>
+        /// Gets a toolbox item from a previously serialized object.
+        /// </summary>
+        /// <param name="serializedObject">serialized Object</param>
+        /// <returns>toolbox item</returns>
+        ToolboxItem DeserializeToolboxItem(object serializedObject, IDesignerHost host);
+
+        /// <summary>
+        /// Get the currently selected tool
+        /// </summary>
+        /// <returns>toolbox item</returns>
+        ToolboxItem GetSelectedToolboxItem();
+
+        /// <summary>
+        /// Get the currently selected tool
+        /// </summary>
+        /// <returns>toolbox item</returns>
+        ToolboxItem GetSelectedToolboxItem(IDesignerHost host);
+
+        /// <summary>
+        /// Gets all .NET Framework tools on the toolbox.
+        /// </summary>
+        /// <returns>toolbox item collection</returns>
+        ToolboxItemCollection GetToolboxItems();
+
+        /// <summary>
+        /// Gets all .NET Framework tools on the toolbox.
+        /// </summary>
+        /// <returns>toolbox item collection</returns>
+        ToolboxItemCollection GetToolboxItems(IDesignerHost host);
+
+        /// <summary>
+        /// Gets all .NET Framework tools on the toolbox fopr specific category.
+        /// </summary>
+        /// <returns>toolbox item collection</returns>
+        ToolboxItemCollection GetToolboxItems(String category);
+
+        /// <summary>
+        /// Gets all .NET Framework tools on the toolbox for specific category.
```

Comment:
Nit: ".NET Framework"... how do we feel about references to ".NET Framework" in this code base?